spec/rspec/matchers/built_in/respond_to_spec.rb

```diff
@@ -72,6 +84,45 @@ def obj.other_method(arg); end
   end
 end
 
+RSpec.describe "expect(...).to respond_to(:new)" do
+  context "with no tampering" do
+    it "will validate new as if it was initialize" do
+      klass = Class.new { def initialize(a, b, c); end }
+      expect(klass).not_to respond_to(:new).with(2).arguments
+      expect(klass).to respond_to(:new).with(3).arguments
+    end
+  end
+
+  context "on a class that has redefined `new`" do
+    it "uses the method signature of the redefined `new` for arg verification" do
+      klass = Class.new { def self.new(a); end }
+      expect(klass).to respond_to(:new).with(1).argument
```

Comment:
It would be good to show a failure for `with(2).arguments` to demonstrate that it is actually using the signature of the redefined `new`; otherwise the implementation could just always pass when it has been redefined.

Reply:
Done